api/client.go

```diff
@@ -366,6 +371,21 @@ func NewClient(c *Config) (*Client, error) {
 	c.modifyLock.Lock()
 	defer c.modifyLock.Unlock()
 
+	// If address begins with a `unix://`, treat it as a socket file path and set
+	// the HttpClient's transport to the corresponding socket dialer.
+	if strings.HasPrefix(c.Address, "unix://") {
+		socketFilePath := strings.TrimPrefix(c.Address, "unix://")
+		c.HttpClient = &http.Client{
```

Comment:
By doing this you're overriding the values set in cleanhttp, plus any values set in DefaultClient, all of which are likely mostly still relevant here. Rather than set an entirely new client and transport, why not simply set a new DialContext on the existing ones?

Reply:
Agreed! Maybe this was the problem I was running into earlier when HTTPS over sockets were throwing weird errors, which had lead me to force TLS disabling on unix sockets for the time being. I've re-enabled that.

Reply:
Addressed in https://github.com/hashicorp/vault/pull/6237/files